### PR TITLE
Code Review Tweaks

### PR DESCRIPTION
**Pull request by billneff79**
- add eslint-config-synacor for static analysis
- fix lint failures
- other small stylistic changes

When you do the diff, enable the "Hide whitespace changes" setting to make it significantly easier to review

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,302 +1,293 @@
...
+	}
+
+	static setClassMappings(mappings) {
+		this._clsMappings = mappings;
+	}
+
+	/**
+	 * @private
+	 */
+	static processMethod(type, data) {
```
Comment (billneff79) [inline index.js:69]: New helper method I created to abstract commonality of all of the static methods below

File: index.js [hunk 2]
```diff
@@ -1,302 +1,293 @@
...
-    return {type:"BasicCard", data:obj};
-  }
-  static Button(obj) {
-    if (this._clsMappings && this._clsMappings['Button']) {
-      return this._clsMappings['Button'](obj);
-    }
-    return {type:"Button", data:obj};
-  }
-
-  static isPromise(obj) {
```
Comment (billneff79) [inline index.js:106]: I pulled this outside of the class since it wasn't class specific.  It's added again below

File: index.js [hunk 3]
```diff
@@ -1,302 +1,293 @@
...
+		if (!convo.conv){
+			if (options.log) {
+				console.log(`${action}:${JSON.stringify(convo, null, 1)}`);
+			}
+			return Promise.resolve(convo);
+		}
+		let finalText = convo._write.reduce((acc, text) => acc.sentence(text), new Say()).toString().trim();
+		let finalSpeech = convo._speak.reduce((acc, text) => acc.sentence(text), new Say()).toString().trim();
+		let promises = [];
+		let textWasPopulated = !!(finalText || finalSpeech);
```
Comment (billneff79) [inline index.js:113]: In javascript, an empty string is falsey, so we don't need the additional variables to hold `textIsNotEmpty` and `speechIsNotEmpty` as long as you trim the strings above.  If you don't want to trim them for some reason, you can undo the change, but given how you are using them, that doesn't seem necessary

File: index.js [hunk 4]
```diff
@@ -1,302 +1,293 @@
...
+			}
+		});
+
+		return isRunningInDialogFlow ? Promise.all(promises) : Promise.all(promises).then(() => convo);
+	}
+
+	/* eslint-enable new-cap */
+	static prepComplete(obj, action = 'close', options) {
+		if (isPromise(obj)) {
+			return obj.then(result => this.prepComplete(result, action, options));
```
Comment (billneff79) [inline index.js:165]: I renamed `obj` to `result` in the arrow function to avoid any ambiguities on which scoped `obj` was being used

File: index.js [hunk 5]
```diff
@@ -1,302 +1,289 @@
...
-  }
-  promise(func) {
-    let r = func(this);
-    if (!Convo.isPromise(r)) {
-      return Promise.resolve(r);
-    }
-    return r;
-  }
-  setAccessToken(token) {
-    if (!conv.user) {
```
Comment (billneff79) [inline index.js:228]: `conv`  is not defined in this scope. I assumed you meant `this.conv`

File: index.js (hunk 3 above)
Comment (tmtek) [reply index.js:113]: That's the Java dev in me not used to "falsey" things =). Thanks for the tip there.

File: index.js (hunk 5 above)
Comment (tmtek) [reply index.js:228]: my bad. Nice catch.
